conn.go

```diff
@@ -52,9 +52,8 @@ type Conn struct {
 	br             *bufio.Reader
 	bw             *bufio.Writer
 
-	readTimeout     chan context.Context
-	writeTimeout    chan context.Context
-	timeoutLoopDone chan struct{}
+	readTimeoutCloser  atomic.Value
+	writeTimeoutCloser atomic.Value
```

Comment:
I think `atomic.Pointer` would be cleaner. `atomic.Value` still stores the value as a pointer, it's just not explicit. The upside of `atomic.Pointer` is that it requires no type casting.